File: src/jniglue/tests/org/OpenColorIO/GpuShaderDescTest.java

import junit.framework.TestCase;
import org.OpenColorIO.*;

public class GpuShaderDescTest extends TestCase {
    
    protected void setUp() {
    }
    
    protected void tearDown() {
    }
    
    public void test_interface() {
        GpuShaderDesc desc = new GpuShaderDesc();

        desc.setFunctionName("foo123");
        assertEquals("foo123", desc.getFunctionName());
        desc.setLut3DEdgeLen(32);
        assertEquals(32, desc.getLut3DEdgeLen());

        desc.setLanguage(GpuLanguage.GPU_LANGUAGE_GLSL_1_0);
        assertEquals(GpuLanguage.GPU_LANGUAGE_GLSL_1_0, desc.getLanguage());
        assertEquals("glsl_1.0 foo123 32", desc.getCacheID());

        desc.setLanguage(GpuLanguage.GPU_LANGUAGE_GLSL_1_3);
        assertEquals(GpuLanguage.GPU_LANGUAGE_GLSL_1_3, desc.getLanguage());
        assertEquals("glsl_1.3 foo123 32", desc.getCacheID());

        desc.setLanguage(GpuLanguage.GPU_LANGUAGE_GLSL_4_0);
        assertEquals(GpuLanguage.GPU_LANGUAGE_GLSL_4_0, desc.getLanguage());
        assertEquals("glsl_4.0 foo123 32", desc.getCacheID());
    }
    
}

File: src/pyglue/tests/GpuShaderDescTest.py

import unittest, os, sys
sys.path.append(os.path.join(sys.argv[1], "src", "pyglue"))
import PyOpenColorIO as OCIO

class GpuShaderDescTest(unittest.TestCase):
    
    def test_interface(self):
        desc = OCIO.GpuShaderDesc()

        desc.setFunctionName("foo123")
        self.assertEqual("foo123", desc.getFunctionName())
        desc.setLut3DEdgeLen(32)
        self.assertEqual(32, desc.getLut3DEdgeLen())

        desc.setLanguage(OCIO.Constants.GPU_LANGUAGE_GLSL_1_0)
        self.assertEqual(OCIO.Constants.GPU_LANGUAGE_GLSL_1_0, desc.getLanguage())
        self.assertEqual("glsl_1.0 foo123 32", desc.getCacheID())

        desc.setLanguage(OCIO.Constants.GPU_LANGUAGE_GLSL_1_3)
        self.assertEqual(OCIO.Constants.GPU_LANGUAGE_GLSL_1_3, desc.getLanguage())
        self.assertEqual("glsl_1.3 foo123 32", desc.getCacheID())

        desc.setLanguage(OCIO.Constants.GPU_LANGUAGE_GLSL_4_0)
        self.assertEqual(OCIO.Constants.GPU_LANGUAGE_GLSL_4_0, desc.getLanguage())
        self.assertEqual("glsl_4.0 foo123 32", desc.getCacheID())

